### PR TITLE
Support Anyscale OpenAI models

### PR DESCRIPTION
**Pull request by mskarlin**
Anyscale supports open source models via a custom base URL in the OpenAI client. This PR enables you to use them in PaperQA by setting the appropriate environment variables and model names. See the docs for Anyscale here: https://docs.anyscale.com/endpoints/text-generation/query-a-model/.

### CODE REVIEW

File: paperqa/docs.py [hunk 1]
```diff
@@ -205,7 +205,15 @@ def set_client(
         embedding_client: Any | None = None,
     ):
         if client is None and isinstance(self.llm_model, OpenAILLMModel):
-            client = AsyncOpenAI()
+            if (api_key := os.environ.get("ANYSCALE_API_KEY")) and (
```
Comment (jamesbraza) [inline paperqa/docs.py:208]: I think LiteLLM does this for us fwiw. So in the future we can consolidate this away

File: tests/test_paperqa.py [hunk 2]
```diff
@@ -544,6 +558,49 @@ def accum(x):
...
+@pytest.mark.asyncio()
+async def test_anyscale_chain():
+    client = AsyncOpenAI(
+        base_url=os.environ["ANYSCALE_BASE_URL"], api_key=os.environ["ANYSCALE_API_KEY"]
+    )
+    llm = OpenAILLMModel(
+        config={
+            "temperature": 0,
+            "model": "meta-llama/Meta-Llama-3-8B-Instruct",
+            "max_tokens": 56,
```
Comment (jamesbraza) [inline tests/test_paperqa.py:574]: Can you document why 56 with a code comment? Or make this set by a formula?

File: tests/test_paperqa.py [hunk 3]
```diff
@@ -544,6 +558,49 @@ def accum(x):
...
+        client,
+        "The {animal} says",
+        skip_system=True,
+    )
+    outputs = []
+
+    def accum(x):
+        outputs.append(x)
+
+    completion = await call({"animal": "duck"}, callbacks=[accum])  # type: ignore[call-arg]
```
Comment (jamesbraza) [inline tests/test_paperqa.py:587]:
```suggestion
    completion = await call({"animal": "duck"}, callbacks=[outputs.append])  # type: ignore[call-arg]
```

**Review by jamesbraza — APPROVED** [review]

File: tests/test_paperqa.py (hunk 2 above)
Comment (mskarlin) [reply tests/test_paperqa.py:574]: it's arbitrary -- other tests use 56, i can do 50 I suppose.

File: paperqa/docs.py (hunk 1 above)
Comment (mskarlin) [reply paperqa/docs.py:208]: for sure

**Pull request merged**